src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Compatibility/Extensions/IndentedStringBuilder.cs

```diff
@@ -125,10 +123,20 @@ public void AppendMultiLineIndented(string text)
 			}
 		}
 
-		public override string ToString()
+		[Obsolete("Consider using ToStringAndFree instead.", error: true)]
+		public new string ToString()
 		{
 			return _stringBuilder.ToString();
 		}
-	}
 
+		public string ToStringAndFree()
+		{
+			var s = _pooledStringBuilder.ToStringAndFree();
+
+			// In case the builder was incorrectly used after a free, let's
+			// fail with a null reference exception instead of having an undefined behavior.
+			_stringBuilder = null!;
```

Comment:
I think it would probably be best to validate for the stringbuilder being null at every use sites. We otherwise risk having to troubleshoot "impossible" conditions.